target/scripts/startup/setup-stack.sh

```diff
@@ -437,10 +437,17 @@ function _setup_ldap
 
   declare -A _dovecot_ldap_mapping
 
+  # Add protocol to LDAP_SERVER_HOST so that we can use dovecot's "uris" option:
+  # https://doc.dovecot.org/configuration_manual/authentication/ldap/
+  dovecot_ldap_server_uri="${LDAP_SERVER_HOST}"
+  if ! echo "${dovecot_ldap_server_uri}" | grep -F '://' >/dev/null; then
```

Comment:
Should be `grep -q` instead of using `>/dev/null`

Reply:
I'm using `grep -F '://'` as your version would replace `ldaps://example.org` with `ldap://ldaps://example.org` - we just need any protocol. the `-F` is for fixed strings (no regex), so I could probably also use a bash comparison here.

Reply:
Would `[[ ${DOVECOT_LDAP_SERVER_URI} != *://* ]]` be fine?

Reply:
Probably a good idea to use a Bash comparison.